spec/models/reporting/patient_visits_per_month_spec.rb

```diff
@@ -0,0 +1,81 @@
+require "rails_helper"
+
+RSpec.describe Reporting::PatientVisitsPerMonth, {type: :model, reporting_spec: true} do
+  describe "Associations" do
+    it { should belong_to(:patient) }
+  end
+
+  describe "the visit definition" do
+    it "considers a BP measurement as a visit" do
+      bp = create(:blood_pressure, :with_encounter, recorded_at: test_times[:now])
+      described_class.refresh
+      with_reporting_time_zones do
+        visit = described_class.find_by(patient_id: bp.patient_id, month_date: test_times[:now])
+        expect(visit.encounter_facility_id).to eq bp.facility_id
+        expect(visit.visited_at).to eq bp.recorded_at
+      end
+    end
+
+    it "considers a Blood Sugar measurement as a visit" do
+      blood_sugar = create(:blood_sugar, :with_encounter, recorded_at: test_times[:now])
+      described_class.refresh
+      with_reporting_time_zones do
+        visit = described_class.find_by(patient_id: blood_sugar.patient_id, month_date: test_times[:now])
+        expect(visit.encounter_facility_id).to eq blood_sugar.facility_id
+        expect(visit.visited_at).to eq blood_sugar.recorded_at
+      end
+    end
+
+    it "considers a Prescription Drug creation as a visit" do
+      prescription_drug = create(:prescription_drug, device_created_at: test_times[:now])
+      described_class.refresh
+      with_reporting_time_zones do
+        visit = described_class.find_by(patient_id: prescription_drug.patient_id, month_date: test_times[:now])
+        expect(visit.visited_at).to eq prescription_drug.device_created_at
```

Comment:
Wondering if we should alias `recorded_at` for these models. Otherwise we have to remember to use date A for this model, and date B for this model.

Reply:
The model itself doesn't expose any other `recorded_at` time, other than `visited_at`. But that's a good idea, to alias the fields for consistency!

Reply:
Fixed in 46d1a40d.